src/spi/traits.rs

```diff
@@ -0,0 +1,155 @@
+/// Helper traits for SPI pins
+use core::ops::Deref;
+use e310x::{qspi0, QSPI0, QSPI1, QSPI2};
+
+#[doc(hidden)]
+pub trait SpiX: Deref<Target = qspi0::RegisterBlock> {}
```

Comment:
Should we seal this trait? https://rust-lang.github.io/api-guidelines/future-proofing.html

Reply:
Good point i'll add that.

Reply:
this is now done